test/renate/memory/test_buffer.py

```diff
@@ -144,16 +199,15 @@ def test_buffer_get_state_dict(max_size):
         metadata = {"x": torch.ones(10, 10)}
         buffer.update(ds, metadata)
     state_dict = buffer.state_dict()
-    for key, value in state_dict["data_points"].items():
-        assert torch.all(torch.eq(buffer._data_points[key], value))
+    torch.all(torch.eq(buffer._data_points[0], state_dict["data_points"][0]))
 
     assert torch.all(torch.eq(buffer.metadata["x"], state_dict["metadata"]["x"]))
 
     for key in ["max_size", "storage_mode", "seed", "count", "data_points"]:
         assert getattr(buffer, "_" + key) == state_dict[key]
     assert buffer.metadata == state_dict["metadata"]
     assert state_dict["size"] == max_size
-    assert len(state_dict["data_points"]["0"]) == max_size
+    assert len(state_dict["data_points"][0]) == max_size
 
 
 @pytest.mark.parametrize("buffer_type", [ReservoirBuffer, SlidingWindowBuffer])
```

Comment:
maybe add independent tests for make_storage, add_data_point etc?

Reply:
There's an (admittedly rudimentary) `test_nested_tensor_storage` above, that tests these methods for different nested tensor structures.